sdks/java/extensions/sql/src/test/java/org/apache/beam/sdk/extensions/sql/impl/parser/BeamDDLTest.java

```diff
@@ -168,6 +174,27 @@ public void testParseDropTable() throws Exception {
     assertNull(tableProvider.getTables().get("person"));
   }
 
+  @Test
+  public void testRegisterSchema() throws Exception {
+    TestTableProvider tableProvider = new TestTableProvider();
+    CalciteConnection connection = JdbcDriver.connect(tableProvider);
+
+    assertEquals("beam", connection.getSchema());
+    assertEquals(
+        Sets.newHashSet("beam", "metadata"), connection.getRootSchema().getSubSchemaNames());
+
+    connection.createStatement().execute("register external schema newSchema type test");
+    assertThat(
+        connection.getRootSchema().getSubSchema("beam").getSubSchemaNames(), contains("newSchema"));
+    SchemaPlus beamSchema = connection.getRootSchema().getSubSchema("beam");
+    SchemaPlus newSchema = beamSchema.getSubSchema("newSchema");
+
+    connection
+        .createStatement()
+        .execute("create table beam.newSchema.someTable (field1 int) type test");
```

Comment:
Why we need to create table from `newSchema` manually?

Does this example still work:
```
CREATE FOREIGN SCHEMA
  TYPE 'bigquery'
  LOCATION 'dataset_example'
  AS bq;

CREATE FOREIGN SCHEMA
  TYPE 'hcatalog'
  LOCATION 'hive-server:2341'
  AS hive;

SELECT * 
  FROM 
       bq.table_example_bq AS bq_table1
     JOIN
       hive.table_example_hive AS hive_table1
     ON
       bq_table1.some_field = hive_table1.some_other_field;
```